### PR TITLE
Improve Loading #3: Drop react-query from async-storage

### PR DESCRIPTION
**Pull request by bruce-glazier**
## Changes

  - Stops using react-query for async-storage to make usage simpler and address some caching/loading issues
  
**Note:** For a better diff of useAsyncStorage review the second commit

For a full description and reasoning behind this: https://github.com/lifeomic/react-native-sdk/pull/358

### CODE REVIEW

File: example/storybook/helpers/oauthConfig.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export const authConfig: AuthConfiguration = {
   additionalParameters: {
     prompt: 'consent',
   },
+  iosPrefersEphemeralSession: true,
```
Comment (bruce-glazier) [inline example/storybook/helpers/oauthConfig.ts:17]: Unrelated change to suppress warning

File: src/hooks/useOnboardingCourse.tsx [hunk 2]
```diff
@@ -36,17 +36,18 @@ export const OnboardingCourseContextProvider = ({
...
+    !!activeProject?.id,
   );
   const [didLaunchCourse, setDidLaunchCourse] = useState<boolean | undefined>(
     undefined,
   );
 
-  const isLoading = isAppConfigLoading || storedDidLaunchResult.isLoading;
-  const isFetched = isAppConfigFetched && storedDidLaunchResult.isFetched;
+  const isLoading = isAppConfigLoading;
+  const isFetched = isAppConfigFetched;
```
Comment (sternetj) [inline src/hooks/useOnboardingCourse.tsx:46]: Should we check to make sure `didLaunchCourse` should be populated here? In theory it seems like the async storage being async could make this hook return `isLoading: false, isFetched: true` before the value is actually loaded.

**Review by sternetj — APPROVED** [review]

File: src/hooks/useOnboardingCourse.tsx (hunk 2 above)
Comment (bruce-glazier) [reply src/hooks/useOnboardingCourse.tsx:46]: Coming up in the next PR.

**Review by sternetj — APPROVED** [review]
